cosmos/config.py

```diff
@@ -54,12 +55,17 @@ class RenderConfig:
     dbt_deps: bool = True
     node_converters: dict[DbtResourceType, Callable[..., Any]] | None = None
     dbt_executable_path: str | Path = get_system_dbt()
-    env_vars: dict[str, str] = field(default_factory=dict)
+    env_vars: dict[str, str] | None = None
     dbt_project_path: InitVar[str | Path | None] = None
 
     project_path: Path | None = field(init=False)
 
     def __post_init__(self, dbt_project_path: str | Path | None) -> None:
+        if self.env_vars:
+            warnings.warn(
+                "RenderConfig.env_vars is deprecated and will be removed in Cosmos 2.0. Use ProjectConfig.env_vars instead.",
```

Comment:
```suggestion
                "RenderConfig.env_vars is deprecated since Cosmos 1.3 and will be removed in Cosmos 2.0. Use ProjectConfig.env_vars instead.",
```
I'm open to this suggestion - but wanted your thoughts on this. It applies to more places.

Until we solve #750, it may be worth being explicit on when we are deprecating things.